Source/diablo.cpp

```diff
@@ -1424,73 +1413,124 @@ void InitKeymapActions()
 	    N_("Open Speedbook."),
 	    'S',
 	    DisplaySpellsKeyPressed,
-	    [&]() { return !IsPlayerDead(); });
+	    nullptr,
+	    CanPlayerTakeAction);
 	sgOptions.Keymapper.AddAction(
 	    "QuickSave",
 	    N_("Quick save"),
 	    N_("Saves the game."),
 	    DVL_VK_F2,
 	    [] { gamemenu_save_game(false); },
-	    [&]() { return !gbIsMultiplayer && !IsPlayerDead(); });
+	    nullptr,
+	    [&]() { return !gbIsMultiplayer && CanPlayerTakeAction(); });
 	sgOptions.Keymapper.AddAction(
 	    "QuickLoad",
 	    N_("Quick load"),
 	    N_("Loads the game."),
 	    DVL_VK_F3,
 	    [] { gamemenu_load_game(false); },
-	    [&]() { return !gbIsMultiplayer && gbValidSaveFile && stextflag == STORE_NONE; });
+	    nullptr,
+	    [&]() { return !gbIsMultiplayer && gbValidSaveFile && stextflag == STORE_NONE && IsGameRunning(); });
 	sgOptions.Keymapper.AddAction(
 	    "QuitGame",
 	    N_("Quit game"),
 	    N_("Closes the game."),
 	    DVL_VK_INVALID,
-	    [] { gamemenu_quit_game(false); });
+	    CanPlayerTakeAction);

```

Comment:
this looks wrong?

Reply:
Copy&Past error. Sorry.
I also changed it (from master) that it also allows quieting when pause is active.

Reply:
>  I also changed it (from master) that it also allows quieting when pause is active.

Small changes like that can go both way, since most of the restrictions comes from the original didn't allow you to access the menu during this state. But that doesn't mean that shortcuts should abide by the same restrictions as they do not depend on the menu state.